aes/aes-soft/src/impls.rs

```diff
@@ -70,7 +71,7 @@ macro_rules! define_aes_impl {
 
             #[inline]
             fn encrypt_block(&self, block: &mut Block) {
-	            let mut blocks = [Block::default(); 2];
+                let mut blocks = [Block::default(); $fixslice_blocks];
```

Comment:
You can use the `FIXSLICE_BLOCKS` constant directly here:

```suggestion
                let mut blocks = [Block::default(); FIXSLICE_BLOCKS];
```